tests/CMakeLists.txt

```diff
@@ -0,0 +1,49 @@
+add_library(common-pg STATIC common-pg.cpp common-pg.hpp common-cleanup.cpp common-cleanup.hpp)
+add_library(middle-tests STATIC middle-tests.cpp middle-tests.hpp)
+
+set(TESTS
+  test-expire-tiles.cpp
+  test-hstore-match-only.cpp
+  test-middle-flat.cpp
+  test-middle-pgsql.cpp
+  test-middle-ram.cpp
+  test-options-database.cpp
+  test-options-parse.cpp
+  test-output-multi-line-storage.cpp
+  test-output-multi-line.cpp
+  test-output-multi-point-multi-table.cpp
+  test-output-multi-point.cpp
+  test-output-multi-poly-trivial.cpp
+  test-output-multi-polygon.cpp
+  test-output-multi-tags.cpp
+  test-output-pgsql-schema.cpp
+  test-output-pgsql-tablespace.cpp
+  test-output-pgsql-z_order.cpp
+  test-output-pgsql.cpp
+  test-parse-xml2.cpp
+  test-pgsql-escape.cpp
+)
+
+
+set(DART_TESTING_TIMEOUT 2000)
+include(Dart)
+
+foreach (test ${TESTS})
+  get_filename_component(test_name ${test} NAME_WE)
+  add_executable(${test_name} ${test})
+  target_link_libraries(${test_name} osm2pgsql_lib common-pg middle-tests)
+  add_test(NAME ${test_name} COMMAND ${test_name}
+    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR})
```

Comment:
Better use ${osm2pgsql_SOURCE_DIR} here to be sure to refer to the correct directory. This is automatically set to the directory where project(osm2pgsql) is called.